chanutils/concurrency.go

```diff
@@ -0,0 +1,20 @@
+package chanutils
+
+import "golang.org/x/sync/errgroup"
+
+// ErrGroup is iterating through values and calls func
+// then waiting for all goroutines to report back. 
+// Returns the first non-nil error (if any).
+func ErrGroup[V any, S []V](f func(V) error, s S) error {
+	eg := &errgroup.Group{}
```

Comment:
See comment below about changing this to be `errgroup.WitContext`.

Reply:
fixed, thanks